fpu/fp.py

```diff
@@ -440,7 +440,7 @@ def reverse(alist):
 
   @staticmethod
   def prepend(alist, e):
-    nlist = CollectionUtils.copy(alist)
+    nlist = list(CollectionUtils.copy(alist))
     nlist.insert(0, e)
     return nlist
```

Comment:
We want to keep container as immutable to meet FP requirement. So possibly, we could do this way:
```python
nlist = CollectionUtils.copy(alist)   # tuple instance
return (e, ) + nlist
```